File: src/types/connection.ts
export type ConnectionMethod = "tcp_ip" | "tcp_ip_over_ssh";

export interface MySQLConfig {
  hostname: string;
  port: number;
  username: string;
  default_schema?: string;
}

export type SSHAuthMethod = "password" | "key_file" | "both";

export interface SSHConfig {
  hostname: string;
  port: number;
  username: string;
  auth_method: SSHAuthMethod;
  key_file_path?: string;
}

export interface Connection {
  id: string;
  name: string;
  folder_id?: string;
  color?: string;
  method: ConnectionMethod;
  mysql: MySQLConfig;
  ssh?: SSHConfig;
  sort_order: number;
  last_connected_at?: string;
  created_at: string;
  updated_at: string;
}

export interface Folder {
  id: string;
  name: string;
  sort_order: number;
  created_at: string;
}

export interface CreateConnectionInput {
  name: string;
  folder_id?: string;
  color?: string;
  method: string;
  mysql_hostname: string;
  mysql_port: number;
  mysql_username: string;
  mysql_password?: string;
  mysql_default_schema?: string;
  ssh_hostname?: string;
  ssh_port?: number;
  ssh_username?: string;
  ssh_password?: string;
  ssh_auth_method?: string;
  ssh_key_file_path?: string;
  ssh_key_passphrase?: string;
}

File: src-tauri/src/models/connection.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Connection {
    pub id: String,
    pub name: String,
    pub folder_id: Option<String>,
    pub color: Option<String>,
    pub method: ConnectionMethod,
    pub mysql: MySQLConfig,
    pub ssh: Option<SSHConfig>,
    pub sort_order: i32,
    pub last_connected_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum ConnectionMethod {
    TcpIp,
    TcpIpOverSsh,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MySQLConfig {
    pub hostname: String,
    pub port: u16,
    pub username: String,
    pub default_schema: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SSHConfig {
    pub hostname: String,
    pub port: u16,
    pub username: String,
    pub auth_method: SSHAuthMethod,
    pub key_file_path: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum SSHAuthMethod {
    Password,
    KeyFile,
    Both,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Folder {
    pub id: String,
    pub name: String,
    pub sort_order: i32,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionWithPassword {
    pub connection: Connection,
    pub mysql_password: Option<String>,
    pub ssh_password: Option<String>,
    pub ssh_key_passphrase: Option<String>,
}
